Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,398 @@
+                    }
+                };
+
+                const AZ::SettingsRegistryInterface::FixedValueString m_viewBookmarksKey;
+                AZStd::unordered_map<AZStd::string, AZStd::vector<ViewBookmark>> m_bookmarkMap;
+                ViewBookmark m_lastKnownLocation;
+            };
+
+            ViewBookmarkComponent* bookmarkComponent = FindBookmarkComponent();
+            if (bookmarkComponent)
+            {
+                // Get the file we want to merge into the settings registry.
+                if (!bookmarkComponent->GetLocalBookmarksFileName().empty())
+                {
+                    auto registry = AZ::SettingsRegistry::Get();
+                    if (registry == nullptr)
+                    {
+                        AZ_Warning(
+                            "SEditorSettings", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
+                        return false;
+                    }
+
+                    auto projectUserRegistryPath = AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Registry/ViewBookmarks";
+                    projectUserRegistryPath /= AZ::IO::FixedMaxPathString(bookmarkComponent->GetLocalBookmarksFileName());
+
+                    // Merge the current viewBookmark file into the settings registry.
+                    bool isMerged = registry->MergeSettingsFile(
+                        projectUserRegistryPath.Native(), AZ::SettingsRegistryInterface::Format::JsonMergePatch, "/O3DE/ViewBookmarks");
+
+                    if (isMerged)
+                    {
+                        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+                        AZStd::string bookmarkKey = s_viewBookmarksRegistryPath + m_bookmarkfileName;
+                        auto viewBookmarkSettingsKey = FixedValueString::format(bookmarkKey.c_str());
+                        ViewBookmarkVisitor viewBookmarkVisitor;
+
+                        const bool visitedViewBookmarks = registry->Visit(viewBookmarkVisitor, viewBookmarkSettingsKey);
+
+                        if (visitedViewBookmarks)
+                        {
+                            m_localBookmarks = viewBookmarkVisitor.m_bookmarkMap.at(m_bookmarkfileName);
+                            m_localBookmarkCount = m_localBookmarks.size() - 1;
+                            m_lastKnownLocation = viewBookmarkVisitor.m_lastKnownLocation;
+                        }
+
+                        // once loaded we can remove the data from the settings registry.
+                        registry->Remove(bookmarkKey + "/");
+                        return visitedViewBookmarks;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        ViewBookmark ViewBookmarkLoader::GetBookmarkAtIndex(int index) const
+        {
+            ViewBookmarkComponent* bookmarkComponent = FindBookmarkComponent();
+            if (bookmarkComponent)
```

Comment:
nit: These can be moved inline inside the if (mentioned before and repeated in a couple of places)